Restore “mark as read” Buttons

mallowigi opened the pull request:
Hi, I've restored the mark as read functionality. I am working with this extension a lot and it annoyed me that such a feature wasn't available so I made my own fork.

I'm not sure what your plans are, you're free to fork from my fork and add your own fixes.

In the meantime I will be using this fork in my everyday.

Thanks!

fregante commented:
Thank you! We can merge this, but can you undo all the unrelated changes? Like the indentation and the changes to manifest.json

fregante commented on extension/github-notifications-preview.css:
```diff
@@ -1,113 +1,114 @@
 .Header .notification-indicator[data-ga-click$=":unread"]::after {
-    display: none;
+  /*display: none;*/
 }
```
If code is not required, it should be dropped, not just commented out

mallowigi replied:
I was unsure why you put display none, so I left it out for your own consideration

mallowigi commented:
I must admit that I didn't test Firefox. Can you do it ?

fregante commented:
I don't see the buttons, but I see this, which isn't great:

<img width="338" src="https://user-images.githubusercontent.com/1402241/87232573-20398880-c3c0-11ea-830d-f03b601765a7.png">

fregante commented on extension/github-notifications-preview.css:
```diff
@@ -123,16 +118,6 @@
 
-.NPG-dropdown .js-notification-action:not(.notification-action-mark-archived) {
-    display: none !important;
-}
```
This change is what's showing the overlapped icons I mentioned

fregante commented on extension/github-notifications-preview.js:
```diff
@@ -105,6 +107,35 @@ async function openDropdown({currentTarget: indicator}) {
 		empty(container);
 		container.append(...list);
 
+		delegate('.NPG-dropdown button', 'click', async event => {
```
This is required in the next version of `delegate-it` coming in #99 

```suggestion
		delegate(document, '.NPG-dropdown button', 'click', async event => {
```

mallowigi commented:
I never developed extensions for Firefox, but I'll check anyways.

mallowigi commented:
By the way, this is how it looks in Chrome: 
![image](https://user-images.githubusercontent.com/5015756/87240396-d1243f80-c421-11ea-852d-5ae76c4b08f0.png)

I'm using the existing behavior of the on hover buttons, I think it's a minor inconvevenience. The problem is that normally the actions are overlapping the timestamp column in the notification, but in the plugin this column is hidden in compact. 

If you want to improve it, please do, but what interests me is the functionality, not the UI. And in that terms I'm okay with my fix.

mallowigi commented:
![image](https://user-images.githubusercontent.com/5015756/87241979-33d10780-c431-11ea-98f9-ea645e1d64ba.png)
I've improved the UI, now I've added ellipsis to the link on hover, so that the buttons don't overlap the text

mallowigi commented:
I've found out that I added the styles from github dark by mistake. I fixed that, but then it means that I'll need to add the styles to the github dark repo as well.

mallowigi commented:
So are you merging ?

mallowigi commented:
anyone home ?

tanmayrajani commented:
Thank you so much, @mallowigi! This is great! Sorry for the delay in merging :)

I noticed a minor improvement with the save/unsave, I'll get that done soon.
